src/bindgen/ir/enumeration.rs

```diff
@@ -897,6 +897,42 @@ impl Source for Enum {
                 out.close_brace(false);
             }
 
+            if config.language == Language::Cxx
+                && config
+                    .enumeration
+                    .derive_tagged_enum_copy_constructor(&self.annotations)
+            {
+                out.new_line();
+                out.new_line();
+                write!(
+                    out,
+                    "{}(const {}& {})",
+                    self.export_name, self.export_name, other
+                );
+                out.new_line();
+                write!(out, " : tag({}.tag)", other);
+                out.open_brace();
+                write!(out, "switch (tag)");
+                out.open_brace();
+                for variant in &self.variants {
+                    if let Some((ref variant_name, ref item)) = variant.body {
+                        write!(
+                            out,
+                            "case {}::{}: new (&{}) {}(other.{}); break;",
+                            self.tag.as_ref().unwrap(),
+                            variant.export_name,
+                            variant_name,
+                            item.export_name(),
+                            variant_name,
+                        );
+                        out.new_line();
+                    }
+                }
+                write!(out, "default: break;");
+                out.close_brace(false);
+                out.close_brace(false);
+            }
```

Comment:
Might be worthwhile to, while here, also generate a `T&&` constructor. 

You'd just need to `std::move` from the passed-in values, you don't need to change the enum variant or anything like that.

Reply:
I think so, though it'd need to be another configuration option (since you may have a non-movable type in the enumeration and you may want a copy-ctor but not a move-ctor, or it may not even compile).

Reply:
(So I think I'll defer it for another PR)